src/components/MultiSelect/MultiSelect.mdx

```diff
@@ -0,0 +1,716 @@
+import { ArgTypes, Canvas, Description, Meta, Source } from "@storybook/blocks";
+
+import { changelogData } from "./multiSelectChangelogData";
+import ComponentChangelogTable from "../../utils/ComponentChangelogTable";
+import * as MultiSelectStories from "./MultiSelect.stories";
+import Link from "../Link/Link";
+
+<Meta of={MultiSelectStories} />
+
+# MultiSelect
+
+| Component Version | DS Version   |
+| ----------------- | ------------ |
+| Added             | `1.4.0`      |
+| Latest            | `Prerelease` |
+
+## Table of Contents
+
+- {<Link href="#overview" target="_self">Overview</Link>}
+- {<Link href="#component-props" target="_self">Component Props</Link>}
+- {<Link href="#accessibility" target="_self">Accessibility</Link>}
+- {<Link href="#populating-the-list-items" target="_self">Populating the List Items</Link>}
+- {<Link href="#visible-list-items" target="_self">Visible List Items</Link>}
+- {<Link href="#nested-list-items" target="_self">Nested List Items</Link>}
+- {<Link href="#disabled-list-items" target="_self">Disabled List Items</Link>}
+- {<Link href="#search-input-field" target="_self">Search Input Field</Link>}
+- {<Link href="#block-element-or-float" target="_self">Block Element or Float</Link>}
+- {<Link href="#width" target="_self">Width</Link>}
+- {<Link href="#default-open-state" target="_self">Default Open State</Link>}
+- {<Link href="#controlling-state-using-selecteditems-and-onchange-prop" target="_self">Controlling state using selectedItems and onChange prop</Link>}
+- {<Link href="#multiselect-nextjs-routing-implementation" target="_self">MultiSelect NextJS routing implementation</Link>}
+- {<Link href="#usemultiselect-hook" target="_self">useMultiSelect hook</Link>}
+- {<Link href="#changelog" target="_self">Changelog</Link>}
+
+## Overview
+
+<Description of={MultiSelectStories} />
+
+## Component Props
+
+The `MultiSelect` renders a non-hierarchical list of items to select from.
+
+<Canvas of={MultiSelectStories.withControls} />
+
+<Source
+  code={`
+<MultiSelect
+  buttonText="MultiSelect",
+  id="multi-select-id"
+  isBlockElement
+  items={[
+    {
+      id: "art",
+      name: "Art",
+    },
+    {
+      id: "architecture",
+      name: "Architecture",
+    },
+    {
+      id: "art_history",
+      name: "Art History",
+    },
+    {
+      id: 'design',
+      name: 'Design'
+    },
+    {
+      id: 'business',
+      name: 'Business'
+    },
+    ...
+  ]}
+/>
+`}
+  language="jsx"
+/>
+
+<ArgTypes of={MultiSelectStories.withControls} />
+
+## Accessibility
+
+The `MultiSelect` component is composed using the DS `Accordion` and `Checkbox`
```

Comment:
I think linking to each storybook doc for Accordion and Checkbox is enough so we don't copy/paste the text. If the text in Accordion or Checkbox gets updated, we'd have to remember to update this  as well.